ledger/apply/application.go

```diff
@@ -464,15 +305,21 @@ func ApplicationCall(ac transactions.ApplicationCallTxnFields, header transactio
 		}
 	}()
 
-	// Keep track of the application ID we're working on
-	appIdx := ac.ApplicationID
-
-	// this is not the case in the current code but still probably better to check
+	// Ensure we are always passed a non-nil ApplyData
 	if ad == nil {
-		err = fmt.Errorf("cannot use empty ApplyData")
+		err = fmt.Errorf("ApplyData cannot be nil")
```

Comment:
Could you make abit more consistent error message ? i.e. 
`ApplyData cannot be nil` vs. `ApplicationCall cannot have nil EvalParams`